src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/Tasks/TaskHelper.cs

```diff
@@ -52,13 +52,10 @@ internal static class TaskHelper
         // </summary>
         internal static void DisplayLogo(TaskLoggingHelper log, string taskName)
         {
-            string acPath = Assembly.GetExecutingAssembly().Location;
-            FileVersionInfo acFileVersionInfo = FileVersionInfo.GetVersionInfo(acPath);
-
-            string avalonFileVersion = acFileVersionInfo.FileVersion;
+            Version avalonFileVersion = Assembly.GetExecutingAssembly().GetName().Version;
```

Comment:
Does Assembly.GetName().Version.ToString() always give the same result as FileVersionInfo.GetVersionInfo(path).FileVersion ?  I doubt it - I couldn't find a guarantee in the docs.

If not, this is a breaking change.  It may not matter here, if the string is only printed out for user's convenience.  But even so, avoid changing the behavior.  Can you use the old code, and only fall back to the new code if you detect the single-file case at runtime?

Reply:
Reading the FileVersion API in runtime I notice that Version is the default product version, so yeah in this case we don't want the version. Although I end up realizing that this is in the PresentationBuildTasks so single file will not affect this code path. I end up reverting the changes to leave it as it was